### DISCUSSION TITLE
Draw a path in full 3D

### DISCUSSION THREAD

**Discussion by thomasl-paintup** [visgl/deck.gl#7960] [Q&A]
Hi, 

I try to draw a PathLayer in full 3D to show the outlines of a building facade or to show the trajectory followed by a robot.
@Pessimistress resolved a bug for the _full3D property on SolidPolygonLayer last month.
Is there a way to implement a similar functionality for the PathLayer.
I was thinking about a parameter that specify the normal vector to the path surface that would be used to calculate the normal to the triangles ?

The actual behaviour is :

- With PathLayer, the lines are not showed if we use [x, y, z] coordinates and are on the YZ plane. BTW, on the documentation all examples are with XY coordinates, but the getPath description is made with XYZ coordinates (except for the example)
- With the LineLayer, the line is always facing the camera is billboard = true, or is facing up (z axis) if billboard = false

I would like, for example, that the line always face x axis from every camera angle.

We are trying to simulate this behaviour by recreating SolidPolygons from Lines as the SolidPolygonLayer work well in full 3D but it's far from ideal.

Thanks for your help.

**Comment by Pessimistress** [comment]
The PathLayer has a billboard option.

**Reply by thomasl-paintup** [reply]
Your right I mixed between LineLayer and PathLayer.
But billboard = true isn't the behaviour we need. We would like the Polygon from the PathLayer to stay normal to the x axis. It's not possible from what I know ?

**Reply by Pessimistress** [reply]
That is not supported right now.
